core/src/main/java/org/jdbi/v3/core/statement/DefineNamedBindingMode.java

```diff
@@ -28,6 +28,9 @@
     /**
      * Bind a boolean {@true} for present values, {@false} for null values.
      * Default behavior.
+     *
+     * Note that due to implementation limitations, null arguments must be instances of
+     * {@link org.jdbi.v3.core.argument.NullArgument} to be recognized.
```

Comment:
People only implementing ArgumentFactories aren't likely to see this though. I think the note should go on the argfac interface and in the devdoc too.

Reply:
But the limitation is specifically related to this strategy only.  It doesn't affect other users.  So shouldn't the note be local to the feature it affects?  The ArgumentFactory interface can't possibly be expected to document its interaction with all other features, otherwise it will be a dumping ground for everything.

Reply:
At least a `caution` in the devdoc (devguide) regarding argumentfactories then. Not just to say "use NullArgument to make feature x and y work", but generally "you really should use our NullArgument to ensure some features keep working".

Reply:
I don't think it's really unfair to have to javadoc it there. Offical features relying on it means it is part of the contract any ArgumentFactory must comply with, so an implementor must know.

Reply:
I added a note to ArgumentFactory.